x/gov/genesis.go

```diff
@@ -90,21 +100,22 @@ func ExportGenesis(ctx sdk.Context, k *keeper.Keeper) (*v1.GenesisState, error)
 	}
 
 	var proposalsDeposits v1.Deposits
-	var proposalsVotes v1.Votes
-	for _, proposal := range proposals {
-		deposits, err := k.GetDeposits(ctx, proposal.Id)
-		if err != nil {
-			return nil, err
-		}
-
-		proposalsDeposits = append(proposalsDeposits, deposits...)
-
-		votes, err := k.GetVotes(ctx, proposal.Id)
-		if err != nil {
-			return nil, err
-		}
+	err = k.Deposits.Walk(ctx, nil, func(_ collections.Pair[uint64, sdk.AccAddress], value v1.Deposit) (stop bool, err error) {
```

Comment:
unrelated, but can we do them each in a go routine?
Maybe the speed-up is irrelevant however 🤷🏾‍♂️

Reply:
Good point, this is something I would love to experiment on as soon as we move modules to use core genesis.